src/Module.ts

```diff
@@ -284,7 +284,11 @@ export default class Module {
 			},
 			isEntry,
 			isExternal: false,
-			meta
+			meta,
+			get syntheticNamedExports() {
+				//since it's public and not readonly, we should use getter to ensure value is always actual
+				return module.syntheticNamedExports;
```

Comment:
Actually I would prefer to follow the route of `hasModuleSideEffects` here which makes this mutable object the source of truth for this property (even though it is public) as it helps reduce internal complexity. I.e. remove `syntheticNamedExports` from `Module` in favour of `info.syntheticNamedExports`. At the very least, it would be consistent.

Reply:
Done!